From 2b1941abb5c20b96ab23bd91b6cc16126c76b4b3 Mon Sep 17 00:00:00 2001
From: David Howells <dhowells@redhat.com>
Date: Tue, 25 Nov 2014 16:36:29 -0500
Subject: [fs] Don't warn if both ->rename() and ->rename2() iops are defined

Message-id: <20141125163629.18115.95262.stgit@warthog.procyon.org.uk>
Patchwork-id: 100947
O-Subject: [PATCH 47/47] Don't warn if both ->rename() and ->rename2() iops are defined
Bugzilla: 985875
RH-Acked-by: Steve Best <sbest@redhat.com>

Don't emit a warning if both the ->rename() and ->rename2() iops are defined.
In the upstream kernel, only one is required, but for the RHEL-7 kernel, I'm
not sure whether there's anything out there that assumes it can use the
->rename() op directly.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/namei.c b/fs/namei.c
index 90e5e65..cb238b7 100644
--- a/fs/namei.c
+++ b/fs/namei.c
@@ -4079,7 +4079,7 @@ int vfs_rename(struct inode *old_dir, struct dentry *old_dentry,
   error = old_dir->i_op->rename(old_dir, old_dentry,
            new_dir, new_dentry);
  } else {
-  WARN_ON(old_dir->i_op->rename != NULL);
+  //WARN_ON(old_dir->i_op->rename != NULL);
   error = rename2(old_dir, old_dentry,
     new_dir, new_dentry, flags);
  }
-- 
1.7.1